### PR TITLE
:book: add beta release issue template to the release tasks docs

### PR DESCRIPTION
**Pull request by nawazkh**
**What this PR does / why we need it**:
- This PR adds beta-release issue-template to the release-tasks docs.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

### CODE REVIEW

File: docs/release/release-tasks.md [hunk 1]
```diff
@@ -406,7 +406,35 @@ We should inform at least the following providers via a new issue on their respe
...
-TODO: Right now we don't have a template for this message but the Comms Team will provide one later. 
+Below mentioned template can be used to update CAPI providers for the beta releases. The template was used to update the CAPI providers after `v1.5.0-beta.0` was released:
+
+```
+<!-- title -->
+# CAPI v1.5.0-beta.0 has been released and is ready for testing 
+<!-- title -->
+
+<!-- body -->
+CAPI `v1.5.0-beta.0` has been released and is ready for testing.
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:417]:
```suggestion

```

File: docs/release/release-tasks.md [hunk 2]
```diff
@@ -406,7 +406,35 @@ We should inform at least the following providers via a new issue on their respe
...
+Below mentioned template can be used to update CAPI providers for the beta releases. The template was used to update the CAPI providers after `v1.5.0-beta.0` was released:
+
+```
+<!-- title -->
+# CAPI v1.5.0-beta.0 has been released and is ready for testing 
+<!-- title -->
+
+<!-- body -->
+CAPI `v1.5.0-beta.0` has been released and is ready for testing. 
+Looking forward to your feedback before CAPI 1.5.0 release, the 25th July 2023!
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:418]:
```suggestion
Looking forward to your feedback before the release on $FULL_RELEASE_DATE!
```

File: docs/release/release-tasks.md [hunk 3]
```diff
@@ -406,7 +406,35 @@ We should inform at least the following providers via a new issue on their respe
  - Packet: https://github.com/kubernetes-sigs/cluster-api-provider-packet/issues/new
  - vSphere: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/new
 
-TODO: Right now we don't have a template for this message but the Comms Team will provide one later. 
+Below mentioned template can be used to update CAPI providers for the beta releases. The template was used to update the CAPI providers after `v1.5.0-beta.0` was released:
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:409]:
```suggestion
Below mentioned template can be used to update CAPI providers for the beta releases. The versions and dates should be updated.
```

File: docs/release/release-tasks.md [hunk 4]
```diff
@@ -406,7 +406,35 @@ We should inform at least the following providers via a new issue on their respe
...
+
+```
+<!-- title -->
+# CAPI v1.5.0-beta.0 has been released and is ready for testing
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:413]:
```suggestion
# CAPI $RELEASE_TAG has been released and is ready for testing 
```

File: docs/release/release-tasks.md [hunk 5]
```diff
@@ -406,7 +406,35 @@ We should inform at least the following providers via a new issue on their respe
...
+<!-- title -->
+# CAPI v1.5.0-beta.0 has been released and is ready for testing 
+<!-- title -->
+
+<!-- body -->
+CAPI `v1.5.0-beta.0` has been released and is ready for testing. 
+Looking forward to your feedback before CAPI 1.5.0 release, the 25th July 2023!
+
+## For quick reference: 
+- [CAPI v1.5.0-beta.0 release notes](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.5.0-beta.0)
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:421]:
```suggestion
- [CAPI $RELEASE_TAG release notes](https://github.com/kubernetes-sigs/cluster-api/releases/tag/$RELEASE_TAG)
```

File: docs/release/release-tasks.md [hunk 6]
```diff
@@ -406,7 +406,35 @@ We should inform at least the following providers via a new issue on their respe
...
+
+## Following are the planned dates for the upcoming releases:
+|Release|Expected Date|
+|-----|-----|
+|v1.5.0-beta.x | Tuesday 5th July 2023|
+|release-1.5 branch created (Begin [Code Freeze])|Tuesday 11th July 2023|
+|v1.5.0-rc.0 released|Tuesday 11th July 2023|
+| release-1.5 jobs created | Tuesday 11th July 2023 |
+| v1.5.0-rc.x released | Tuesday 18th July 2023 |
+| v1.5.0 released	| Tuesday 25th July 2023 |
```
Comment (killianmuldoon) [inline docs/release/release-tasks.md:433]: TODO: Insert release calendar from `docs/release/releases/release-XXXX`

Comment (killianmuldoon) [inline docs/release/release-tasks.md:433]:
```suggestion
TODO: Insert updated release calendar from `docs/release/releases/release-XXXX`

|-----|-----|
|v1.5.0-beta.x | Tuesday 5th July 2023|
|release-1.5 branch created (Begin [Code Freeze])|Tuesday 11th July 2023|
|v1.5.0-rc.0 released|Tuesday 11th July 2023|
| release-1.5 jobs created | Tuesday 11th July 2023 |
| v1.5.0-rc.x released | Tuesday 18th July 2023 |
| v1.5.0 released	| Tuesday 25th July 2023 |
```

**Review by killianmuldoon — COMMENTED** [review]
Thanks for adding this! 

I think it should be made it a bit more generic to highlight places where the versions etc. should be replaced. 

We should add an item in the release improvements backlog to automate this in some way.

File: docs/release/release-tasks.md [hunk 7]
```diff
@@ -406,7 +406,35 @@ We should inform at least the following providers via a new issue on their respe
  - Packet: https://github.com/kubernetes-sigs/cluster-api-provider-packet/issues/new
  - vSphere: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/new
```
Comment (sbueringer) [inline docs/release/release-tasks.md:408]: Q: Should we start putting stuff like this in a separate file? We could have docs/release/communication/file and then just link it from here

Comment (killianmuldoon) [reply docs/release/release-tasks.md:408]: IMO it's fine to have here for now as a first step. This process should ideally be automated so it's not a load of copy-paste work though.

File: docs/release/release-tasks.md (hunk 6 above)
Comment (nawazkh) [reply docs/release/release-tasks.md:412]: Updated via aa868c3690288209fa0366854b483e85b2945093

Comment (nawazkh) [reply docs/release/release-tasks.md:412]: Updated via aa868c3690288209fa0366854b483e85b2945093

File: docs/release/release-tasks.md (hunk 7 above)
Comment (nawazkh) [reply docs/release/release-tasks.md:388]: Keeping it as is for now :)

**Review by killianmuldoon — COMMENTED** [review]
/lgtm

**Review by killianmuldoon — COMMENTED** [review]
@nawazkh do you think we can close this in favor of https://github.com/kubernetes-sigs/cluster-api/pull/9110?

**Pull request closed without merging**